### PR TITLE
Updating to fit with fetch aborting

### PR DESCRIPTION
**Pull request by jakearchibald**
***
[Preview](https://s3.amazonaws.com/pr-preview/jakearchibald/xhr/fetch-abort.html) | [Diff](https://s3.amazonaws.com/pr-preview/whatwg/xhr/7e332a5...jakearchibald:bbe3e77.html)

### CODE REVIEW

File: xhr.bs [hunk 1]
```diff
@@ -526,7 +524,7 @@ methods, when invoked, must run these steps:
   <p>Set variables associated with the object as follows:
 
   <ul>
-   <li><p>Unset the <a><code>send()</code> flag</a>, <a>stop timeout flag</a>, and
+   <li><p>Unset the <a><code>send()</code> flag</a>, and
```
Comment (annevk) [inline xhr.bs]: No need for a comma anymore.

File: xhr.bs [hunk 2]
```diff
@@ -893,8 +893,8 @@ method must run these steps:
        <li><p>while {{XMLHttpRequest/timeout!!attribute}} attribute value is not zero.
       </ol>
 
-     <li><p>If <var>req</var>'s <a for=request>done flag</a> is unset, then
-     <a for=fetch>terminate</a> <a for=/>fetching</a> with reason <i>timeout</i>.
+     <li><p>If <var>req</var>'s <a for=request>done flag</a> is unset, then set the
+     <a>timed out flag</a>, and <a for=fetch>terminate</a> <a for=/>fetching</a>.
```
Comment (annevk) [inline xhr.bs]: Also no need for a comma here.

File: xhr.bs [hunk 3]
```diff
@@ -1042,12 +1039,9 @@ method must run these steps:
     <p>Let <var>response</var> be the result of
     <a for=/>fetching</a> <var>req</var>.
 
-    <p>If the {{XMLHttpRequest/timeout!!attribute}} attribute value is not
-    zero, <a for=fetch>terminate</a> this
-    <a for=/>fetch</a> if it has not returned
-    within the amount of milliseconds from the
-    {{XMLHttpRequest/timeout!!attribute}} attribute value with reason
-    <i>timeout</i>.
+    <p>If the {{XMLHttpRequest/timeout!!attribute}} attribute value is not zero, set the
```
Comment (annevk) [inline xhr.bs]: then set* if we're touching this

File: xhr.bs [hunk 4]
```diff
@@ -1042,12 +1039,9 @@ method must run these steps:
...
+    <a>timed out flag</a> and <a for=fetch>terminate</a> <a for=/>fetching</a> if it has not
+    returned within the amount of milliseconds from the {{XMLHttpRequest/timeout!!attribute}}.
```
Comment (annevk) [inline xhr.bs]: I wonder if we should clarify what returned means here. But maybe as a follow-up since that's not your fault.

File: xhr.bs [hunk 5]
```diff
@@ -418,7 +416,7 @@ turn causes the <a>use-CORS-preflight flag</a> to be set.)
 <p>To <dfn>terminate the request</dfn>,
 <a for=fetch>terminate</a> the
 <a for=/>fetch</a> algorithm operated by the
-{{XMLHttpRequest}} object with reason <i>fatal</i>.
+{{XMLHttpRequest}} object.
```
Comment (annevk) [inline xhr.bs]: Maybe inline this algorithm altogether at this point given you're removing 1 of the 3 callsites?

**Comment by annevk** [comment]
@yutakahirano could you please review this as well? Thanks!

File: xhr.bs (hunk 2 above)
Comment (jakearchibald) [reply xhr.bs]: Is the rule "use commas if there are more than two things to do"?

Comment (annevk) [reply xhr.bs]: Yeah, or if it's needed because it would get unwieldy somehow.

File: xhr.bs [hunk 6]
```diff
@@ -1122,33 +1110,24 @@ method must run these steps:
...
+ <li><p>If the <a>timed out flag</a> is set, then run the <a>request error steps</a> for event
+ <a event><code>timeout</code></a> and exception <code>TimeoutError</code>.
 
-  <dl class=switch>
-   <dt><i>end-user abort</i>
-   <dd><p>Run the <a>request error steps</a> for event
-   <a event><code>abort</code></a> and exception
-   <code>AbortError</code>.
+ <li>
+  <p>Otherwise, if <var>response</var>'s <a for=response>body</a>'s <a for=body>stream</a> has
```
Comment (yutakahirano) [inline xhr.bs]: is errored

**Comment by annevk** [comment]
Proposed commit message:
```
Align with Fetch's upcoming abort revamp

See https://github.com/whatwg/fetch/pull/523 for details.
```

**Comment by jakearchibald** [comment]
I'm happy with that.

**Pull request merged**